teamcity: failed tests on master: acceptance/TestDockerCLI/test_client_side_checking.tcl

The following tests appear to have failed:

[#40628](https://teamcity.cockroachdb.com/viewLog.html?buildId=40628):

```
--- FAIL: acceptance/TestDockerCLI/test_client_side_checking.tcl (114.416s)
	replication_test.go:46: replication test is a no-op for empty cluster

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/cli/interactive_tests/test_sql_mem_monitor.tcl
+++ b/pkg/cli/interactive_tests/test_sql_mem_monitor.tcl
@@ -68,7 +68,12 @@ send "select * from columns as a, columns as b, columns as c, columns as d limit
 
 # Check that the query crashed the server
 set spawn_id $shell_spawn_id
-eexpect "out of memory"
+# Error is either "out of memory" (Go) or "cannot allocate memory" (C++)
+expect {
+    "out of memory" {}
+    "cannot allocate memory" {}
+    timeout {exit 1}
+}
 eexpect ":/# "
 
 # Check that the client got a bad connection error

